pkg/writer.go

```diff
@@ -53,14 +57,45 @@ type Writer interface {
 	Close() (err error)
 }
 
+type writeManyOptions struct {
+	concurrency   int
+	writeCallback func(uint32)
+}
+
+type WriteManyOption func(options *writeManyOptions)
+
+func WithConcurrency(concurrency int) WriteManyOption {
+	return func(options *writeManyOptions) {
+		options.concurrency = concurrency
+	}
+}
```

Comment:
we need to figure out what to do with a malformed values here, what does 0 mean? what do values < 0 mean?  Do we treat them as numcpu or 1?